Client/Frontend/Browser/Playlist/BasicAuthCredentials.swift

```diff
@@ -0,0 +1,86 @@
+// Copyright 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import UIKit
+
+class BasicAuthCredentialsManager: NSObject, URLSessionDataDelegate {
```

Comment:
Are we implementing this just for the staging server? Isn't the playlist folder sharing links supposed to be on https://playlist.brave.com/, which doesn't require authentication?

Reply:
we could remove it at this point imo,
staging endpoints hid behind basic auth, and thats not the case anymore

Reply:
I can remove this now. However, I would like to keep this code for future features that are hidden behind basic auth.
Just the class itself, even if not used.

The reason is because we are duplicating the same code and doing the same thing because it wasn't flexible enough before:

https://github.com/brave/brave-ios/blob/65f165a370cdaf56ec8d383272f8f7b8dca17feb/Client/Frontend/Browser/Authenticator.swift#L49

https://github.com/brave/brave-ios/blob/6f1f414b2a3ea4110c3caeb6b6f27a9e664bdc8a/Client/Frontend/Browser/Search/BraveSearchManager.swift#L255

So in a separate PR, I want to refactor all the other code to just use the standard `BasicAuthCredentialsManager` and get rid of all the duplicates. They're doing the same thing, but in different places.

Reply:
I'm fine to keep it. Just wanted to make sure we don't actually rely on basic auth in production and it's there for the development purposes